tests/topologyDescription/topologyDescription-debug-001.phpt

```diff
@@ -0,0 +1,40 @@
+--TEST--
+MongoDB\Driver\TopologyDescription debug output
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = create_test_manager();
+
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    public function topologyChanged(MongoDB\Driver\Monitoring\TopologyChangedEvent $event)
+    {
+        $topologyDescription = $event->getNewDescription();
+        var_dump($topologyDescription);
+    }
+}
+
+$subscriber = new MySubscriber;
+MongoDB\Driver\Monitoring\addSubscriber($subscriber);
+
+$command = new MongoDB\Driver\Command(['ping' => 1]);
+$manager->executeCommand(DATABASE_NAME, $command);
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+object(MongoDB\Driver\TopologyDescription)#%d (%d) {
+  %a
+}
+object(MongoDB\Driver\TopologyDescription)#%d (%d) {
+  %a
+}
+object(MongoDB\Driver\TopologyDescription)#%d (%d) {
+  %a
+}
+===DONE===
```

Comment:
I wrote the expected outputs for all the tests according to the results I got from running them, but I don't really understand why each test results in three outputs and am not sure whether I should make the expected outputs more specific.

Reply:
We can take a look at the output later, but it's natural for there to be multiple TopologyChanged events emitted. That occurs in many of the existing [SDAM Monitoring specs tests](https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring/tests/monitoring).

In this particular test, the only purpose of the event subscriber is to provide a TopologyDescription object. We don't actually care about what or how many events are dispatched. The subscriber can be changed to only capture a single TopologyDescription object. Some other points to consider:

 * The add/removeSubscriber functions can also be called from anywhere.
 * You can use Manager methods instead of global functions. That would actually be make more sense, since those methods were created with SDAM Monitoring in mind (see: [PHPC-1666](https://jira.mongodb.org/browse/PHPC-1666)).
 * The subscriber is an object, so you can structure it as you like with properties and extra methods if you like. There is no requirement to do everything from within the event handling method. If you look through PHPC tests for existing CommandSubscriber implementations, you'll see a few different variations.

Reply:
I updated the tests to use the `Manager::addSubscriber()` method. How would I change the subscriber to only capture a single TopologyDescription object? I think the tests are failing on the GitHub CI (https://github.com/mongodb/mongo-php-driver/pull/1244/checks?check_run_id=3263852984) right now because only two TopologyChanged events are emitted, but when I run locally I get three.

Reply:
Sorry, missed this last comment in my GitHub notifications.

> How would I change the subscriber to only capture a single TopologyDescription object?

We know the `topologyChanged()` _method_ is going to be invoked for each TopologyChangedEvent emitted by libmongoc's SDAM layer; however you control the implementation of the `topologyChanged()` method and have full control over what to do (or not do) with the event object(s) you receive.